pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -185,7 +186,7 @@ class ModelInterpreter implements Serializable {
                 if (skippedReason == null && parent != null &&
                     root.hasSatisfiedConditions(parent.post, script.getProperty("currentBuild"), parent, firstError)) {
```

Comment:
This call passes `Stage parent` as `Object context` to Root.hasSatisfiedConditions, from where the context propagates to BuildCondition.getCombinedResult, which however expects the context to be the name of the stage as a String, rather than a Stage instance. Filed [JENKINS-68288](https://issues.jenkins.io/browse/JENKINS-68288).